solution-1/balaram/task-3.py

```diff
@@ -0,0 +1,37 @@
+# Problem: Write a recursive program to flatten a nested dictionary. The dictionary can be nested to any levels.
+
+def flatten_dict(dictionary: dict, parent_key = '') -> dict:
+    """ """
+    separator = "_"
+    items = []
+    for key, value in dictionary.items():
+        if parent_key:
+            new_key = parent_key + separator + key
+        else:
+            new_key = key
+        
+        if type(value) == type({}):
+            items.extend(flatten_dict(value, new_key).items())
+        else:
+            items.append((new_key, value))
+    return dict(items)
+
+
+
+nested_dict = {
+    'a': 1,
+    'b': {
+        'c': 2,
+        'd': {
+            'e': 3,
+            'f': {
+                'g': 4
+            }
+        }
+    },
+    'h': 5
+}
```

Comment:
try running for 
```
nested_dict = {
    'a': 1,
    'b': {
        'c': 2,
        'd': {
            'a': 3,
            'f': {
                'b': 4
            }
        }
    },
    'f': 5
}
```

Reply:
this test cases passes without the suggested change as well. I can't find the need for using parent_key when I'm already using new_key which is made using parent_key. Am I missing something here?